sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/rest_client.hpp

```diff
@@ -2159,6 +2170,26 @@ namespace Azure { namespace Storage { namespace Blobs {
         Models::CopyStatus CopyStatus;
       };
     } // namespace _detail
+    /**
+     * @brief Optional, default 'replace'.  Indicates if source tags should be copied or replaced
+     * with the tags specified by x-ms-tags.
+     */
+    class BlobCopySourceTagsMode final {
```

Comment:
Use an `ExtendableEnumeration` here - that allows you to get rid of a LOT of boilerplate.

Reply:
The code gen was written before we have the base class. Apparently we haven't got time to adapt the extendable enum base class. Tracked in this issue. https://github.com/Azure/autorest.cpp/issues/109

Reply:
Ah - I was unaware that rest_client was autogenerated.